File: providers/openrouter.js
module.exports = {
  
  name: 'openrouter',
  aliases: ['or'],
  displayName: 'OpenRouter',
  description: 'OpenRouter — multi-model API aggregator',
  baseUrl: 'https://openrouter.ai/api/v1',
  apiMode: 'openai_chat',
  envVars: ['OPENROUTER_API_KEY', 'OPENROUTER_BASE_URL'],
  authType: 'api_key',
  defaultModel: 'auto',
  defaultMaxTokens: 4096,
  signupUrl: 'https://openrouter.ai',
  supportsHealthCheck: true,
  defaultHeaders: { 'HTTP-Referer': 'https://github.com/robert/august-proxy', 'X-Title': 'August Proxy' },
  modelProfiles: {
    '*': { supportsReasoning: false, supportsThinking: false, combinedBudget: false, contextWindow: 131072, maxOutputTokens: 4096 },
  },

};

File: backend-py/app/providers/openrouter.py
"""
Provider config for OpenRouter.
"""
from __future__ import annotations
INFO: dict[str, object] = {'name': 'OpenRouter', 'aliases': ['or'], 'display_name': 'OpenRouter', 'description': 'OpenRouter — multi-model API aggregator', 'base_url': 'https://openrouter.ai/api/v1', 'api_mode': 'openai_chat', 'env_vars': ['OPENROUTER_API_KEY', 'OPENROUTER_BASE_URL'], 'auth_type': 'api_key', 'default_model': 'auto', 'default_max_tokens': 4096, 'default_headers': {'HTTP-Referer': 'https://github.com/robert/august-proxy', 'X-Title': 'August Proxy'}, 'signup_url': 'https://openrouter.ai', 'supports_health_check': True, 'model_profiles': {'*': {'supportsReasoning': False, 'supportsThinking': False, 'combinedBudget': False, 'contextWindow': 131072, 'maxOutputTokens': 4096}}}

def resolveBaseUrl() -> str:
    return 'https://openrouter.ai/api/v1'

def resolveApiKey(envKey: str | None=None) -> str | None:
    return envKey